scheduler-worker/src/main/java/gov/nasa/jpl/aerie/scheduler/worker/services/SynchronousSchedulerAgent.java

```diff
@@ -439,6 +444,16 @@ private PlanComponents loadInitialPlan(final PlanMetadata planMetadata, final Pr
         } else {
           throw new Error("Unhandled variant of DurationType:" + schedulerActType.getDurationType());
         }
+        //look for durations in initial sim results
+        if(initialSimulationResults.isPresent()){
+          for(final var simAct: initialSimulationResults.get().simulatedActivities.entrySet()){
+            if(simAct.getValue().directiveId().isPresent() &&
+               simAct.getValue().directiveId().get().equals(elem.getKey())){
+              actDuration = simAct.getValue().duration();
+            }
+          }
+        }
+
```

Comment:
Should this block only be in the `// Do nothing` cases of Uncontrollable, Fixed, and Parametric? If the activity is Controllabe, its duration would've already been set on L432.

Additionally, is it possible to set the Duration of Fixed/Parametric parameters during this method?

Reply:
> Should this block only be in the // Do nothing cases of Uncontrollable, Fixed, and Parametric? If the activity is Controllabe, its duration would've already been set on L432.
It is not really related to duration type and I did not want to make it look like it was. But why not.

> Additionally, is it possible to set the Duration of Fixed/Parametric activities during this method?
Absolutely, let's make it smarter